my-app/src/utils/useWindowResize.tsx

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react';
+
+export const useWindowResize = () => {
+    const [width, setWidth] = useState(window.innerWidth);
+    const [height, setHeight] = useState(window.innerHeight);
+
+    const listener = () => {
+        setWidth(window.innerWidth);
+        setHeight(window.innerHeight);
+    }
+
+    useEffect(() => {
+        window.addEventListener('resize', listener);
```

Comment:
In real-world application it's good to use `debounce` here to prevent excessive function call